### ISSUE TITLE
kotlinx-coroutines-debug version 1.6.3 breaks build with Spring Boot Upgrade to 2.7.1

### ISSUE THREAD

**Issue by jesperancinha** [Kotlin/kotlinx.coroutines#3345]
Hi there!

Yesterday I noticed that on several of my repos the build was breaking and dependabot could not automate any of the Spring Boot updates. After an exhaustive investigation I came accross for `kotlinx-coroutines-debug` which is included in spring boot 2.7.1, there is jar dependency configured for the build of materials and that of course generates:

```shell
[ERROR] Failed to execute goal on project stamps-and-coins-service: Could not resolve dependencies for project <> org.jetbrains.kotlinx:kotlinx-coroutines-bom:jar:1.6.3 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

This is what I see when looking into the `kotlinx-coroutines-debug` dependency:

```xml
 <dependencies>
    <dependency>
      <groupId>org.jetbrains.kotlinx</groupId>
      <artifactId>kotlinx-coroutines-core</artifactId>
      <version>1.6.3</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.jetbrains.kotlinx</groupId>
      <artifactId>kotlinx-coroutines-bom</artifactId>
      <version>1.6.3</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>net.java.dev.jna</groupId>
      <artifactId>jna</artifactId>
      <version>5.9.0</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>net.java.dev.jna</groupId>
      <artifactId>jna-platform</artifactId>
      <version>5.9.0</version>
      <scope>runtime</scope>
    </dependency>
    <dependency>
      <groupId>org.jetbrains.kotlin</groupId>
      <artifactId>kotlin-stdlib-jdk8</artifactId>
      <version>1.6.21</version>
      <scope>runtime</scope>
    </dependency>
  </dependencies>
```

So I'm guessing the problem is here:

```xml
    <dependency>
      <groupId>org.jetbrains.kotlinx</groupId>
      <artifactId>kotlinx-coroutines-bom</artifactId>
      <version>1.6.3</version>
      <scope>runtime</scope>
    </dependency>
```

Are you planning on making a release soon about this?

Thanks!

**Comment by dkhalanskyjb** [comment]
Duplicate of https://github.com/Kotlin/kotlinx.coroutines/issues/3334

**Comment by jesperancinha** [comment]
@dkhalanskyjb , It is a duplicated but the issue was reported in 1.6.2. Will this be fixed in 1.6.4?

**Comment by dkhalanskyjb** [comment]
It was only reported after we released 1.6.3. Also, sure, we'll fix this.

**Comment by MrTomato** [comment]
It works if you exclude the bom dependency and then add it manually to the pom :)

```
        <dependency>
            <groupId>io.kotest</groupId>
            <artifactId>kotest-runner-junit5-jvm</artifactId>
            <version>5.3.2</version>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <groupId>org.jetbrains.kotlinx</groupId>
                    <artifactId>kotlinx-coroutines-bom</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlinx</groupId>
            <artifactId>kotlinx-coroutines-bom</artifactId>
            <version>1.6.3</version>
            <type>pom</type>
            <scope>test</scope>
        </dependency>
```
